server/protocols/webadmin/webadmin-cassandra-data/src/test/java/org/apache/james/webadmin/routes/CassandraMappingsRoutesTest.java

```diff
@@ -0,0 +1,222 @@
+        webAdminServer.await();
+
+        RestAssured.requestSpecification = WebAdminUtils.buildRequestSpecification(webAdminServer)
+            .setBasePath(CassandraMappingsRoutes.ROOT_PATH)
+            .build();
+    }
+
+    @After
+    public void tearDown() {
+        webAdminServer.destroy();
+        taskManager.stop();
+        cassandra.clearTables();
+    }
+
+    @AfterClass
+    public static void tearDownClass() {
+        cassandra.closeCluster();
+    }
+
+    @Test
+    public void postMappingsActionWithSolvedInconsistenciesQueryParamShouldCreateATask() {
+        given()
+            .queryParam("action", MAPPINGS_ACTION)
+        .when()
+            .post()
+        .then()
+            .statusCode(HttpStatus.CREATED_201)
+            .header("Location", is(notNullValue()))
+            .body("taskId", is(notNullValue()));
+    }
+
+    @Test
+    public void postMappingsActionWithSolvedInconsistenciesQueryParamShouldHaveSuccessfulCompletedTask() {
+        String taskId = with()
+            .queryParam("action", MAPPINGS_ACTION)
+            .post()
+            .jsonPath()
+            .get("taskId");
+
+        given()
+            .basePath(TasksRoutes.BASE)
+        .when()
+            .get(taskId + "/await")
+        .then()
+            .body("status", is("completed"))
+            .body("taskId", is(notNullValue()))
+            .body("type", is(CassandraMappingsSolveInconsistenciesTask.TYPE))
+            .body("startedDate", is(notNullValue()))
+            .body("submitDate", is(notNullValue()))
+            .body("completedDate", is(notNullValue()));
+    }
+
+    @Test
+    public void postMappingsActionShouldRejectInvalidActions() {
+        given()
+            .queryParam("action", "invalid-action")
+        .when()
+            .post()
+        .then()
+            .statusCode(HttpStatus.BAD_REQUEST_400)
+            .body("statusCode", is(400))
+            .body("type", is(ErrorResponder.ErrorType.INVALID_ARGUMENT.getType()))
+            .body("message", is("Invalid arguments supplied in the user request"))
+            .body("details", is("No enum constant org.apache.james.webadmin.dto.ActionMappings.invalid-action"));
+    }
+
+    @Test
+    public void postMappingsActionShouldRequireAction() {
+        when()
+            .post()
+        .then()
+            .statusCode(HttpStatus.BAD_REQUEST_400)
+            .body("statusCode", is(400))
+            .body("type", is(ErrorResponder.ErrorType.INVALID_ARGUMENT.getType()))
+            .body("message", is("Invalid arguments supplied in the user request"))
+            .body("details", is("'action' url parameter is mandatory"));
+    }
+
+    @Test
+    public void postMappingsActionShouldResolveRRTInconsistencies() {
+        cassandraRecipientRewriteTableDAO.addMapping(SOURCE_1, MAPPING);
+        cassandraRecipientRewriteTableDAO.addMapping(SOURCE_2, MAPPING);
+
+        cassandraMappingsSourcesDAO.addMapping(MAPPING, SOURCE_1);
```

Comment:
could you have an assertion to show that data is inconsistency? Then it makes conviction at your final assertion that data inconsistency is resolved

Reply:
I imagine 
```java
assertThat(cassandraRecipientRewriteTableDAO.mappingSources(MAPPING))
       .isNotEqualTo(cassandraMappingsSourcesDAO.mappingSources(MAPPING));
```

Reply:
I believe that there is enough tests in `CassandraMappingsSourcesDAO` and `CassandraRecipientRewriteTableDAO` to be sure that we have the inconsistency present here